### PR TITLE
Handle zeek-inet-ntop snprintf() return value correctly

### PR DESCRIPTION
**Pull request by timwoj**

### CODE REVIEW

**Comment by timwoj** [comment]
I would also be fine with converting this code to c++ so that we can use `std::string` as well, if only so we don't have to deal with all of this.

File: zeek_inet_ntop.c [hunk 1]
```diff
@@ -174,21 +174,39 @@ zeek_inet_ntop6(const u_char *src, char *dst, socklen_t size)
 			remaining -= strlen(tp);
 			break;
 		}
-		tp += snprintf(tp, remaining, "%x", words[i]);
+
+		// snprintf() returns the number of characters that were written not
+		// including the null character. We can use that to increase the
+		// pointer as we're moving forward. Unfortunately, sprintf() can also
```
Comment (awelzel) [inline zeek_inet_ntop.c]:
```suggestion
		// pointer as we're moving forward. Unfortunately, snprintf() can also
```

File: zeek_inet_ntop.c [hunk 2]
```diff
@@ -149,7 +149,7 @@ zeek_inet_ntop6(const u_char *src, char *dst, socklen_t size)
 	 */
 	int remaining = sizeof(tmp);
 	tp = tmp;
-	for (i = 0; i < (NS_IN6ADDRSZ / NS_INT16SZ); i++) {
+	for (i = 0; i < (NS_IN6ADDRSZ / NS_INT16SZ) && remaining > 0; i++) {
```
Comment (awelzel) [inline zeek_inet_ntop.c:152]: Is line 165 below missing a `remaining--` ? Can't comment there.
https://github.com/zeek/zeek-3rdparty/pull/23/files#diff-c9e934660269e2c4174cbae9201b0c174dda7489eda32b772aba401893e3cc0bR165

**Review by awelzel — COMMENTED** [review]
Phew - this is all subtle.

Is there any immediate plan to ditch `zeek_inet_ntop.c`, otherwise maybe a doctest.h based testhere would be useful to give a bit more confidence?

File: zeek_inet_ntop.c (hunk 2 above)
Comment (timwoj) [reply zeek_inet_ntop.c:152]: Yep, fixed.

**Review by awelzel — APPROVED** [review]
Assuming CI passes in the Zeek repo.

**Pull request merged**